### ISSUE TITLE
Some sort of middleware (kind of)

### ISSUE THREAD

**Issue by ItsVoltz** [maxence-charriere/go-app#673]
Hello so i'm trying to work on some sort of "middleware" to include in all of my routes that require you to be logged in. However I cannot get it to properly write the data to the component.

"middleware":
```go
func authenticated(ctx app.Context, user *responses.User) {
	ctx.Async(func() {
		// fetch if auth token
		var auth string
		err := ctx.SessionStorage().Get("auth", &auth)

		// not authorized
		if err != nil || auth == "" {
			app.Log("not authorized", auth, err)
			ctx.Navigate("/login")
			return // not authorized
		}

		response, err := api.User(auth)
		if err != nil {
			app.Log("failed to fetch user", err)
			ctx.Navigate("/login")
			return // not authorized
		}

		user = response
	})
}
```

How im using it in my code:
```go
func (c *dashboard) OnNav(ctx app.Context) {
	authenticated(ctx, &c.User)
}
```

I have tried a few different ways such as calling comp.Update(), but cannot get it to actually set the value. Any idea how I could properly set the value? thru my "middleware"

**Comment by maxence-charriere** [comment]
Where do you use this?

**Comment by maxence-charriere** [comment]
Where do you set de session storage kv ?

**Comment by ItsVoltz** [comment]
> Where do you set de session storage kv ?

Session store from when the client gets authenticated successfully. Then when I made the "middleware" I want to grab user info from the API and pass it into a user response object. Which the authenticated response works fine when I do it hardcoded into each page and use dispatch. (Note yes I have tried using ctx.Dispatch to also set the value from my authenticated func
